subprojects/base-services/src/main/java/org/gradle/internal/resources/AbstractResourceLockRegistry.java

```diff
@@ -80,11 +80,19 @@ public void execute(ResourceLock resourceLock) {
         return new Action<ResourceLock>() {
             @Override
             public void execute(ResourceLock resourceLock) {
-                threadResourceLockMap.remove(Thread.currentThread().getId(), resourceLock);
+                unassociateWorkerLease(resourceLock);
             }
         };
     }
 
+    public void associateWorkerLease(ResourceLock sharedResource) {
```

Comment:
❌ Let's not put "workerLease" in the name of these methods as this class is used for all types of `ResourceLock`.  Let's call this `associateResourceLock` or `useResourceLock` or something like that.